f600233d1b154d507b782f84604bcb12/update/sys_script_include_d393df0197141110c351f5b3f153af5d.xml

```diff
@@ -8,22 +8,106 @@
+			xmlDoc.parseXML(xmlStringBody);
+			var jsonObject = gs.xmlToJSON(xmlDoc);
+
+			// As a result of parseXML, we are left with LineItems as an Array or an Object
+			// If it is an Array, that means there are CLIN records to create
+			// If it is an Object, that means there is a single CLIN record instead of a CLIN and INFOSLIN, requiring no CLIN creation
+			var lineItems = jsonObject.ProcurementDocument.AwardInstrument.ContractLineItems.LineItems;
+			var clins = [];
+			gs.info(Array.isArray(lineItems));
+			if (Array.isArray(lineItems)){
+				for (var i = 0; i< lineItems.length; i++){
+					// If LineItemType is INFOSLIN, we will create a CLIN record
+					if (lineItems[i].LineItemIdentifier.DFARS.LineItem.LineItemType == "INFOSLIN"){
+						var lineItemBase = lineItems[i].LineItemIdentifier.DFARS.LineItem.LineItemBase;
+						var popStartDate, popEndDate;
+						// We need to loop through lineItems again to find the matching CLIN that contains the popStartDate and popEndDate
+						for (var j = 0; j< lineItems.length; j++){
+							var lineItemToMatch = lineItems[j].LineItemIdentifier.DFARS.LineItem.LineItemBase;
+							if (lineItems[j].LineItemIdentifier.DFARS.LineItem.LineItemType == "CLIN" & lineItemToMatch == lineItemBase & lineItems[j].hasOwnProperty("LineItemDates")){
+								popStartDate = lineItems[j].LineItemDates.LineItemPeriod.PeriodStart.DateElement;
+								popEndDate = lineItems[j].LineItemDates.LineItemPeriod.PeriodEnd.DateElement;
+							}
+						}
+						var obligatedAmount = lineItems[i].LineItemAmounts.ItemObligatedAmount.ObligatedAmount;
+						// Create new CLIN record
+						var grClin = new GlideRecord("x_g_dis_atat_clin");
+						grClin.initialize();
+						grClin.setValue("clin_number", lineItemBase);
+						grClin.setValue("funds_obligated", obligatedAmount);
+						grClin.setValue("pop_start_date", popStartDate);
+						grClin.setValue("pop_end_date", popEndDate);
+						var newCLINSysID = grClin.insert();
+						clins.push(newCLINSysID);
+					}
+				}
+			}
+			// Check for created CLIN(s)
+			if (clins.length > 0){
+				// Extract startDate and endDate for the new Task Order record
+				var startDate = jsonObject.ProcurementDocument.AwardInstrument.ProcurementInstrumentHeader.ProcurementInstrumentDates.ProcurementInstrumentPeriods.StartDate;
+				var endDate = jsonObject.ProcurementDocument.AwardInstrument.ProcurementInstrumentHeader.ProcurementInstrumentDates.ProcurementInstrumentPeriods.EndDate;
+				// Create new Task Order parent record
+				var grTaskOrder = new GlideRecord("x_g_dis_atat_task_order");
+				grTaskOrder.initialize();
+				// W911S122F0003 is the hardcoded value, this will need to be altered later
+				grTaskOrder.setValue("task_order_number", "W911S122F0003");
+				grTaskOrder.setValue("pop_start_date", startDate);
+				grTaskOrder.setValue("pop_end_date", endDate);
+				grTaskOrder.setValue("clins", clins.toString());
+				var newTaskOrderSysID = grTaskOrder.insert();
```

Comment:
Was `newTaskOrderSysID` supposed to be used somewhere else?

Reply:
I instantiated this variable so it could be used in logging when I was testing, I can remove the variable and just do:
```
grTaskOrder.insert();
```
instead, because we aren't using it for logging anymore.